### PR TITLE
feat: ForumChannel default_reaction_emoji attribute

### PR DESCRIPTION
**Pull request by davidhozic**
## Summary
This implements #2174.

## Information
Added ``default_reaction_emoji`` parameter to ``Guild.create_forum_channel()`` + ``ForumChannel.edit()`` and ``default_reaction_emoji`` attribute to ``ForumChannel``.

Example:

```py
import discord
import secret

client = discord.Client()

@client.listen("on_ready")
async def start():
    guild = client.get_guild(863071397207212052)

    channel = await guild.create_forum_channel("Test", default_reaction_emoji="😍")
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.delete()

    channel = await guild.create_forum_channel("Test", default_reaction_emoji=863077925757911070)
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.delete()

    channel = await guild.create_forum_channel("Test", default_reaction_emoji=client.get_emoji(863077925757911070))
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.delete()

    channel = await guild.create_forum_channel("Test", default_reaction_emoji="<:coffee_break:863077925757911070>")
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.delete()

    channel = await guild.create_forum_channel("Test", default_reaction_emoji="<a:coffee_break:863077925757911070>")
    print(f"Emoji is: {channel.default_reaction_emoji}")

    await channel.edit(default_reaction_emoji=863077925757911070)
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.edit(default_reaction_emoji=client.get_emoji(863077925757911070))
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.edit(default_reaction_emoji="😙")
    print(f"Emoji is: {channel.default_reaction_emoji}")
    await channel.edit(default_reaction_emoji="<coffee_break:863077925757911070>")
    print(f"Emoji is: {channel.default_reaction_emoji}")

    await channel.delete()

client.run(secret.TOKEN)
```

- [ ] This PR fixes an issue.
- [x] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting,
      examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why.
- [x] I have updated the changelog to include these changes.

### CODE REVIEW

**Review by plun1331 — DISMISSED** [review]
`edit` and `create` methods could accept a string representation of the emoji (`<a:emoji_name:emoji_id>`) for convenience, similarly to how `Message.add_reaction` accepts a string repr of a custom emoji (although it's implemented differently due to how the endpoint works)

other than that lgtm

**Comment by davidhozic** [comment]
> `edit` and `create` methods could accept a string representation of the emoji (`<a:emoji_name:emoji_id>`) for convenience, similarly to how `Message.add_reaction` accepts a string repr of a custom emoji (although it's implemented differently due to how the endpoint works)
> 
> other than that lgtm

It does accept ``<a:emoji_name:emoji_id>``, I'll update the docstrings to make it clearer.

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]
Fix documentation formatting

See [here](https://pycord--2178.org.readthedocs.build/en/2178/api/models.html#discord.Guild.create_forum_channel), for example.

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,9 @@ These changes are available on the `master` branch, but have not yet been releas
 
 ### Added
 
+- `Guild.create_forum_channel` new parameter and `ForumChannel` new attribute:
+  `default_reaction_emoji`.
+  ([#2178](https://github.com/Pycord-Development/pycord/pull/2178))
```
Comment (JustaSqu1d) [inline CHANGELOG.md:17]: Entries should be added in chronological order.

**Review by JustaSqu1d — COMMENTED** [review]
Move the changelog entry to the bottom

**Comment by plun1331** [comment]
> > `edit` and `create` methods could accept a string representation of the emoji (`<a:emoji_name:emoji_id>`) for convenience, similarly to how `Message.add_reaction` accepts a string repr of a custom emoji (although it's implemented differently due to how the endpoint works)
> > other than that lgtm
> 
> It does accept `<a:emoji_name:emoji_id>`, I'll update the docstrings to make it clearer.

It didn't look like it did at a first glance, sorry about that

**Review by plun1331 — DISMISSED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -72,6 +72,9 @@ These changes are available on the `master` branch, but have not yet been releas
 - Added `suppress` and `allowed_mentions` parameters to `Webhook` and
   `InteractionResponse` edit methods.
   ([#2138](https://github.com/Pycord-Development/pycord/pull/2138))
+- Added `Guild.create_forum_channel` parameter and `ForumChannel` attribute:
+  `default_reaction_emoji`.
```
Comment (JustaSqu1d) [inline CHANGELOG.md:76]:
```suggestion
- Added `default_reaction_emoji` parameter to `Guild.create_forum_channel()`, `ForumChannel.create_forum_channel()`, and
`ForumChannel.edit_forum_channel()` functions:
```

**Review by JustaSqu1d — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (JustaSqu1d) [reply CHANGELOG.md:76]: Probably also add a separate changelog entry about the `default_reaction_emoji` attribute.

File: CHANGELOG.md [hunk 3]
```diff
@@ -72,6 +72,11 @@ These changes are available on the `master` branch, but have not yet been releas
 - Added `suppress` and `allowed_mentions` parameters to `Webhook` and
   `InteractionResponse` edit methods.
   ([#2138](https://github.com/Pycord-Development/pycord/pull/2138))
+- Added `default_reaction_emoji` parameter to `Guild.create_forum_channel()` and
+  `ForumChannel.edit()` methods.
+  ([#2178](https://github.com/Pycord-Development/pycord/pull/2178))
+- Added `ForumChannel.default_reaction_emoji` attribute.
+  ([#2178](https://github.com/Pycord-Development/pycord/pull/2178))
```
Comment (Lulalaby) [inline CHANGELOG.md:79]:
```suggestion
- Added `ForumChannel.default_reaction_emoji` attribute.
  ([#2178](https://github.com/Pycord-Development/pycord/pull/2178))
- Added `default_reaction_emoji` parameter to `Guild.create_forum_channel()` and
  `ForumChannel.edit()` methods.
  ([#2178](https://github.com/Pycord-Development/pycord/pull/2178))
```

**Review by JustaSqu1d — COMMENTED** [review]
What does this do

https://github.com/davidhozic/pycord/blob/feat/default_forum_reaction/discord/abc.py#L511-L514

**Comment by davidhozic** [comment]
> What does this do

Which thing, I don't see any reference in your reply?

**Comment by davidhozic** [comment]
> What does this do
> 
> https://github.com/davidhozic/pycord/blob/feat/default_forum_reaction/discord/abc.py#L511-L514

It checks if the key exists - if parameter was given to edit. If it doesn't it will raise an Exception and not execute the else statement

**Review by plun1331 — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Pull request merged**